site/content/en/docs/concepts/topology_aware_scheduling.md

```diff
@@ -0,0 +1,169 @@
+date: 2024-04-11
+weight: 6
+description: >
+  Allows scheduling of Pods based on the topology of nodes in a data center.
+---
+
+{{< feature-state state="alpha" for_version="v0.9" >}}
+
+It is common that AI/ML workloads require a significant amount of pod-to-pod
+communication. Therefore the network bandwidth between the running Pods
+translates into the workload execution time, and the cost of running
+such workloads. The available bandwidth between the Pods depends on the placement
+of the Nodes, running the Pods, in the data center.
+
+We observe that the data centers have a hierarchical structure of their
+organizational units, like racks and blocks, where there are multiple nodes
+within a rack, and there are multiple racks within a block. Pods running within
+the same organizational unit have better network bandwidth than Pods on
+different units. We say that nods placed in different racks are more distant
+than nodes placed within the same rack. Similarly, nodes placed in different
+blocks are more distant than two nodes within the same block.
+
+In this feature (called Topology Aware Scheduling, or TAS for short) we
+introduce a convention to represent the
+[hierarchical node topology information](#node-topology-information), and a set
+of APIs for Kueue administrators and users to utilize the information
+to optimize the Pod placement.
+
+### Node topology information
+
+We propose a lightweight model for representing the hierarchy of nodes within a
+data center by using node labels. In this model the node labels are set up by a
+cloud provider, or set up manually by administrators of on-premise clusters.
+
+Additionally, we assume that every node used for TAS has a set of the labels
+which identifies uniquely its location in the tree structure. We do not assume
+global uniqueness of labels on each level, i.e. there could be two nodes with
+the same "rack" label, but in different "blocks".
+
+For example, this is a representation of the dataset hierarchy;
+
+|  node  |  cloud.provider.com/topology-block | cloud.provider.com/topology-rack |
+|:------:|:----------------------------------:|:--------------------------------:|
+| node-1 |               block-1              |              rack-1              |
+| node-2 |               block-1              |              rack-2              |
+| node-3 |               block-2              |              rack-1              |
+| node-4 |               block-2              |              rack-3              |
+
+Note that, there is a pair of nodes, node-1 and node-3, with the same value of
+the "cloud.provider.com/topology-rack" label, but in different blocks.
+
+### Admin-facing APIs
+
+As an admin, in order to enable the feature you need to:
+1. ensure the `TopologyAwareScheduling` feature gate is enabled
+2. create at least one instance of the `Topology` API
+3. reference the `Topology` API from a dedicated ResourceFlavor by the
+   `.spec.topologyName` field
+
+#### Example
+
+```yaml
```

Comment:
Could we move this to example folder?

https://github.com/kubernetes-sigs/kueue/tree/main/site/static/examples

Reply:
done